sys/syslog/backend.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Continental
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_syslog
+ *
+ * @brief       Syslog backend definitions
+ *
+ * @{
+ * @file
+ *
+ * @author      Vincent Dupont <vincent.dupont@continental-its.com>
+ */
+
+#ifndef SYSLOG_BACKEND_H
+#define SYSLOG_BACKEND_H
+
+#include <stdint.h>
+#include <stdatomic.h>
+
+#include "thread.h"
+#include "time.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef CONFIG_SYSLOG_MAX_STRING_LEN
+#define CONFIG_SYSLOG_MAX_STRING_LEN    256
+#endif
+
+struct syslog_msg {
+    /* metadata */
+    uint8_t pri;
+    struct tm time;
+    bool time_set;
+    const char *app_name;
+    kernel_pid_t proc_id;
+    /* msg */
+    char msg[CONFIG_SYSLOG_MAX_STRING_LEN];
+    size_t len;
+    /* internal */
+    atomic_int cnt;
+};
+
+/**
+ * @brief   syslog backend driver
+ */
+typedef struct {
+    /**
+     * @brief   syslog backend init
+     */
+    int (*init)(void);
```

Comment:
Can't we get rid of this function pointer and just initialize the backend within `auto_init` one by one?

Reply:
yes, I guess we could. I initially thought it would be handy to have this, but I don't see a lot of added value.